tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardNumberMapper.java

```diff
@@ -42,6 +42,63 @@ public class CardNumberMapper {
 		GeneralMapper.class
 	);
 
+	@Named("unknownToNull")
+	protected String mapUnknownToNull(final String value) {
+		final String mappedValue = generalMapper.mapBlankStringToNull(value);
+		if (mappedValue != null && mappedValue.contains("?")) {
+			return null;
+		}
+		return mappedValue;
+	}
+
+	@Mapping(
+		target = "stringValue",
+		source = "stringValue",
+		qualifiedByName = "unknownToNull"
+	)
+	@Mapping(
+		target = "setCode",
+		source = "setCode",
+		qualifiedByName = "unknownToNull"
+	)
+	@Mapping(
+		target = "printNumberPrefix",
+		source = "printNumberPrefix",
+		qualifiedByName = "unknownToNull"
+	)
+	@Mapping(
+		target = "printNumberSuffix",
+		source = "printNumberSuffix",
+		qualifiedByName = "unknownToNull"
+	)
+	protected abstract CardNumber copyWithoutUnknowns(
+		final CardNumber cardNumber
+	);
+
+	@Mapping(target = "printNumber", ignore = true)
+	protected abstract CardNumber mapWithUnknownPrintNumber(
+		final CardNumber cardNumber
+	);
+
+	@AfterMapping
+	protected CardNumber updateCardNumber(
+		final CardNumber originalCardNumber,
+		@MappingTarget final CardNumber updatedCardNumber
+	) {
+		if (originalCardNumber.equals(updatedCardNumber)) {
+			return originalCardNumber;
+		}
+		// this indicates an unknown print/set
+		if (updatedCardNumber.stringValue() == null) {
+			// TODO: mark somehow the CardNumber
```

Comment:
For this, we can include a `boolean` on the `CardNumber` to mark as unreleased

Reply:
I think that it is enough to just remove this TODO: removing the `stringValue` should be enough to recognize unreleased prints, in addition to not having a proper ID.